python/mxnet/gluon/block.py

```diff
@@ -510,15 +524,26 @@ def load_parameters(self, filename, ctx=None, allow_missing=False,
 
         if not any('.' in i for i in loaded.keys()):
             # legacy loading
-            del loaded
+            loaded = None
+            # We cannot `del loaded` as it is used in a generator expression
+            # below. `del loaded` would be a SyntaxError in Python 2.
```

Comment:
this comment is confusing as it comments on code that doesn't exist.

in general, comments should be minimal, and concise, and only out of necessity. code should be self-documenting.

Reply:
It's non-obvious that `del loaded` is a SyntaxError in Python 2. But I explained the explanation as it is confusing. Rather let's just instruct future readers to replace the instruction with `del loaded` once Python 2 support is gone.